src/main/kotlin/lotto/domain/LotteryStatistician.kt

```diff
@@ -0,0 +1,29 @@
+package lotto.domain
+
+class LotteryStatistician(
+    targetLottoStr: String,
+) {
+    private val targetLotto = Lotto(targetLottoStr.split(", ").map { it.toInt() })
+
+    fun statistics(lotties: List<Lotto>): WinningStatistics {
+        val statistics = initStatistics()
+
+        lotties
+            .groupingBy { LottoRank.of(it.matchCount(targetLotto)) }
+            .eachCount()
+            .forEach { (rank, count) -> rank?.let { statistics[it] = count } }
+
+        return WinningStatistics(
+            purchaseAmount = lotties.size * Lotto.AMOUNT_PER_LOTTO,
+            statistics = statistics,
+        )
+    }
+
+    private fun initStatistics(): MutableMap<LottoRank, Int> =
+        LottoRank.entries.associateWith { 0 }.toMutableMap()
```

Comment:
열거타입을 Key로 사용하는 맵 자료구조라면 EnumMap을 활용해보는건 어떨까요? 

```suggestion
    private fun initStatistics(): MutableMap<LottoRank, Int> =
    EnumMap(LottoRank::class.java).apply {
        LottoRank.entries.forEach { this[it] = 0 }
    }
```

Reply:
[a44d275](https://github.com/next-step/kotlin-lotto/pull/1024/commits/a44d27531f4681a0b66d13c1bd2ad502f403c2d4)